sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelUtils.java

```diff
@@ -51,18 +62,117 @@
    *
    * <p>For example, when reading from "gs://bucket/path", the scheme "gs" is
```

Comment:
{@link PipelineOptions} are required to provide dependencies and pipeline level configuration to the individual {@link IOChannelFactory}s.